Fix broken inspec test

This is blocking PRs from getting reviewed properly and merged. Affected:

* https://github.com/ansible/molecule/pull/2034
* https://github.com/ansible/molecule/pull/2033
* https://github.com/ansible/molecule/pull/2036

I cannot reproduce this locally. I ran:

```
$ tox -e py37-ansible27-functional -- -v -k test_command_init_role_inspec -p no:warning
```

Log is (not useful at all ... another case against `sh`?):

From https://travis-ci.com/ansible/molecule/jobs/199661714#L451.
```console
=================================== FAILURES ===================================
________________________ test_command_init_role_inspec _________________________
temp_dir = local('/tmp/pytest-of-travis/pytest-0/test_command_init_role_inspec0/SVPPY')
    @needs_inspec
    @needs_rubocop
    def test_command_init_role_inspec(temp_dir):
        role_directory = os.path.join(temp_dir.strpath, 'test-init')
        options = {
            'role_name': 'test-init',
            'verifier_name': 'inspec',
        }
        cmd = sh.molecule.bake('init', 'role', **options)
        pytest.helpers.run_command(cmd)
        pytest.helpers.metadata_lint_update(role_directory)
    
        with change_dir_to(role_directory):
>           sh.molecule('test')
/home/travis/build/ansible/molecule/test/functional/docker/test_scenarios.py:104: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/travis/build/ansible/molecule/.tox/ansible27-functional/lib/python3.7/site-packages/sh.py:1427: in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
/home/travis/build/ansible/molecule/.tox/ansible27-functional/lib/python3.7/site-packages/sh.py:774: in __init__
    self.wait()
/home/travis/build/ansible/molecule/.tox/ansible27-functional/lib/python3.7/site-packages/sh.py:792: in wait
    self.handle_command_exit_code(exit_code)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = --> Validating schema /tmp/pytest-of-travis/pytest-0/test_command_init_role_inspec0/SVPPY/test-init/molecule/defa...changed=1    unreachable=0    failed=0
    
--> Pruning extra files from scenario ephemeral directory
code = 1
    def handle_command_exit_code(self, code):
        """ here we determine if we had an exception, or an error code that we
        weren't expecting to see.  if we did, we create and raise an exception
        """
        ca = self.call_args
        exc_class = get_exc_exit_code_would_raise(code, ca["ok_code"],
                ca["piped"])
        if exc_class:
            exc = exc_class(self.ran, self.process.stdout, self.process.stderr,
                    ca["truncate_exc"])
>           raise exc
E           sh.ErrorReturnCode_1: 
E           
E             RAN: /home/travis/build/ansible/molecule/.tox/ansible27-functional/bin/molecule test
E           
E             STDOUT:
E           --> Validating schema /tmp/pytest-of-travis/pytest-0/test_command_init_role_inspec0/SVPPY/test-init/molecule/default/molecule.yml.
E           Validation completed successfully.
E           --> Test matrix
E               
E           └── default
E               ├── lint
E               ├── dependency
E               ├── cleanup
E               ├── destroy
E               ├── syntax
E               ├── create
E               ├── prepare
E               ├── converge
E               ├── idempotence
E               ├── side_effect
E               ├── verify
E               ├── cleanup
E               └── destroy
E               
E           --> Scenario: 'default'
E           --> Action: 'lint'
E           --> Executing Yamllint on files found in /tmp/pytest-of-travis/pytest-0/test_command_init_role_inspec0/SVPPY/test-init/...
E           Lint completed successf... (1530 more, please see e.stdout)
E           
E             STDERR:
/home/travis/build/ansible/molecule/.tox/ansible27-functional/lib/python3.7/site-packages/sh.py:815: ErrorReturnCode_1
----------------------------- Captured stdout call -----------------------------
    --> Initializing new role test-init...
    Initialized role in /tmp/pytest-of-travis/pytest-0/test_command_init_role_inspec0/SVPPY/test-init successfully.
```

--- a/molecule/cookiecutter/scenario/verifier/inspec/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/{{cookiecutter.verifier_directory}}/test_default.rb
+++ b/molecule/cookiecutter/scenario/verifier/inspec/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/{{cookiecutter.verifier_directory}}/test_default.rb
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 # Molecule managed
 
 describe file('/etc/hosts') do

--- a/test/conftest.py
+++ b/test/conftest.py
@@ -40,6 +40,9 @@ def run_command(cmd, env=os.environ, log=True):
     if log:
         cmd = _rebake_command(cmd, env)
 
+    # Never let sh truncate exceptions in testing
+    cmd = cmd.bake(_truncate_exc=False)
+
     return util.run_command(cmd)
 
 

--- a/test/functional/docker/test_scenarios.py
+++ b/test/functional/docker/test_scenarios.py
@@ -101,7 +101,8 @@ def test_command_init_role_inspec(temp_dir):
     pytest.helpers.metadata_lint_update(role_directory)
 
     with change_dir_to(role_directory):
-        sh.molecule('test')
+        cmd = sh.molecule.bake('test')
+        pytest.helpers.run_command(cmd)
 
 
 def test_command_init_scenario_inspec(temp_dir):
@@ -138,7 +139,8 @@ def test_command_init_role_goss(temp_dir):
     pytest.helpers.metadata_lint_update(role_directory)
 
     with change_dir_to(role_directory):
-        sh.molecule('test')
+        cmd = sh.molecule.bake('test')
+        pytest.helpers.run_command(cmd)
 
 
 def test_command_init_scenario_goss(temp_dir):
@@ -254,7 +256,8 @@ def test_command_init_role_with_template(temp_dir):
     pytest.helpers.metadata_lint_update(role_directory)
 
     with change_dir_to(role_directory):
-        sh.molecule('test')
+        cmd = sh.molecule.bake('test')
+        pytest.helpers.run_command(cmd)
 
 
 @pytest.mark.parametrize(

